### DISCUSSION TITLE
Plan for a reference data schema? sigma support?

### DISCUSSION THREAD

**Discussion by juju4** [openobserve/openobserve#5136] [Ideas]
To standardize and normalized queries, it would be very useful to have a fixed data model independent of ingestion mode.
Especially for security SIEM, all providers have such a schema. An existing one (elasticsearch?) would probably simplify things.

* Elastic Common Schema (ECS) Reference https://www.elastic.co/guide/en/ecs/current/index.html
* Common Information Model Add-on Manual https://docs.splunk.com/Documentation/CIM/6.0.0/User/Overview
* https://github.com/SumoLogic/cloud-siem-content-catalog/blob/master/schema/full_schema.md
* Advanced Security Information Model (ASIM) https://learn.microsoft.com/en-us/azure/sentinel/normalization

Along that, an official support of https://github.com/SigmaHQ/sigma with converter tool (https://github.com/SigmaHQ/pySigma) support would be great. It would allow easy use of a full repository of queries.

Thanks

**Comment by prabhatsharma** [comment]
You could use VRL today to normalize schema. We don't provide a prebuilt VRL functions list today that help you do it, but the functionality exists for you to achieve normalization fairly easily and use ECS/OCSF/OTEL semantic conventions or anything else. An example of the VRL script to normalize IP address  could be:

```js
.ip = .ip_address
del(.ip_address)
```

**Comment by juju4** [comment]
From https://openobserve.ai/docs/user-guide/functions/#what-are-functions, VRL functions can apply both at ingestion and search. But I would think this remapping to be resource-costly if done on both side. Or possible?
That clearly helps but after, normalizing, is ideally with same schema for everyone. No plan for such reference on openobserve side?

**Reply by prabhatsharma** [reply]
If you have vrl function at ingestion time then you don't need it at query time.

**Reply by juju4** [reply]
It was more where it makes more sense, probably better at ingestion if not doing too many else will impact performance. right?
does VRL can apply between different streams?
if I ingest same type of data, through filebeat in one environment and in another with otel-collector, how do I merge them together and have the same structure?
